pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,172 @@
+                        for _, ctr := range pr.Containers {
+                                for _, dev := range ctr.Devices {
+                                        podDeviceIDs += fmt.Sprint(dev.DeviceIds)
+                                }
+                                if len(podDeviceIDs) > 2 {
+                                        // example: [0000:81:02.0 0000:81:0a.3]
+                                        // remove "[]" and return DeviceIds
+                                        length := len(podDeviceIDs) - 1
+                                        podDeviceIDs = podDeviceIDs[1:length]
+                                        klog.V(2).Info("pod DeviceIds of ", podName, podNamespace, "are: ", podDeviceIDs)
+                                        return podDeviceIDs, nil
+                                }
+                        }
+                }
+        }
+
+        return "", nil
+}
+
+func GetPodPCIDeviceIDRequest(args *skel.CmdArgs) string {
+        // Parse pod resources list and get the PCI device for K8S_POD_NAME and K8S_POD_NAMESPACE
+        os.Setenv("POD_DEVICE_ID", "")
+        k8sArgs, err := GetK8sArgs(args)
+        if k8sArgs != nil && err == nil {
+                deviceID, err := GetPodResourceRequests(string(k8sArgs.K8S_POD_NAME), string(k8sArgs.K8S_POD_NAMESPACE))
+                if deviceID != "" && err == nil {
+                       os.Setenv("POD_DEVICE_ID", deviceID)
+               }
+        }
+       return os.Getenv("POD_DEVICE_ID")
+}
+
+// GetVFInfo takes in a VF's PCI device id and returns its PF and VF ID.
+func GetVFInfo(vfPCI string) (string, int, error) {
+        var vfID int
+
+        pf, err := sriovcniutils.GetPfName(vfPCI)
+        if err != nil {
+                return "", vfID, err
+        }
+
+        vfID, err = sriovcniutils.GetVfid(vfPCI, pf)
+        if err != nil {
+                return "", vfID, err
+        }
+
+        return pf, vfID, nil
+}
+
+// GetVFLinkNames returns VF's network interface name given it's PCI address.
+func GetVFLinkNames(pciAddress string) (string, error) {
+        return sriovcniutils.GetVFLinkNames(pciAddress)
+}
+
+// configureSecondaryInterface adds Secondary Interface support.
+// Will also support SR-IOV interfaces.
+func (pc *podConfigurator) configureSecondaryInterface(
+        podName string,
+        podNameSpace string,
+        containerID string,
+        containerNetNS string,
+        containerIFDev string,
+        mtu int,
+        sriovVFDeviceID string,
+        result *current.Result,
+	isSriovPodLink bool,
+) error {
+        err := pc.ifConfigurator.configureContainerLink(podName, podNameSpace, containerID, containerNetNS, containerIFDev, mtu, sriovVFDeviceID, result, isSriovPodLink)
+        if err != nil {
+                return err
+        }
+        hostIface := result.Interfaces[0]
+        containerIface := result.Interfaces[1]
+
+        if sriovVFDeviceID != "" {
```

Comment:
As so far we support only SRIOV interface, what you think we check: sriovVFDeviceID != "" and isSriovPodLink == true, at the begining of the func (like 156) and return an error if not the case?

Reply:
Add update the comments on the func about the limitation (only SRIOV interface is supported as of now).

Reply:
Done